### ISSUE TITLE
GRPS INSECURE CONDITION - Argo-cd Helm chart Chart ingress-grpc.yaml

### ISSUE THREAD

**Issue by orenr2301** [argoproj/argo-helm#2496]
### Describe the bug

In ingress-grpc yaml file template

{{- if and .Values.server.ingressGrpc.enabled (eq .Values.server.ingress.controller "generic") -}}
{{- $hostname := .Values.server.ingressGrpc.hostname | default (printf "grpc.%s" .Values.server.ingress.hostname) -}}
{{- $insecure := index .Values.configs.params "server.insecure" | toString -}}
**`{{- $servicePort := eq $insecure "true" | ternary .Values.server.service.servicePortHttp .Values.server.service.servicePortHttps -}}`**
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: {{ include "argo-cd.server.fullname" . }}-grpc
  namespace: {{ .Release.Namespace | quote }}
  
  SHOULD BE: 
  
  **`{{- $servicePort := eq $insecure "true" | ternary .Values.server.service.servicePortHttps .Values.server.service.servicePortHttp -}}`**
  
  Till not fixed the server is not injecting the right port makes the server insecure even when you put it in a false. 
  
  I don't know what you are smoking, but I want too XD

### Related helm chart

argo-cd

### Helm chart version

6.0.2

### To Reproduce

1. deploy the chart with server-insecure key with **true** value
2. try to access argocd URL - make sure you have certificates and all that is needed 
3. Check what port your service is getting 

Basically, you will see your error and you should understand it quickly 

### Expected behavior

you should be getting errors when trying to access the ARGO-CD URL if you are keeping the current config as they are now

**Comment by orenr2301** [comment]
You Welcome, glad to help

**Comment by pdrastil** [comment]
Resolved via https://github.com/argoproj/argo-helm/pull/2497
